src/test/fuzz/txgraph.cpp

```diff
@@ -138,19 +138,18 @@ struct SimTxGraph
     }
 
     /** Add a new transaction to the simulation. */
-    TxGraph::Ref* AddTransaction(const FeePerWeight& feerate)
+    void AddTransaction(TxGraph::Ref&& ref, const FeePerWeight& feerate)
```

Comment:
nit: the change breaks simmetry with https://github.com/bitcoin/bitcoin/blob/48d4b936e09f82a3bcb65fe5850977e0e526e634/src/txgraph.cpp#L750 but simplifies the public interface of `TxGraph`.

Q: would it make sense to migrate https://github.com/bitcoin/bitcoin/blob/aef40b93cf057d2a27d61881b0858d491206bcd3/src/txgraph.cpp#L2142-L2175 to `void TxGraphImpl::AddTransaction(TxGraph::Ref&& ref, const FeePerWeight& feerate) noexcept` as well which might enable us deleting `TxGraph::Ref::Ref(Ref&& other) noexcept` as well?
Most of its usages seem to be in tests.